unittests/resources/checks/frontend_checks.py

```diff
@@ -63,10 +63,15 @@ class NoPrgEnvCheck(BaseFrontendCheck):
 
 @rfm.simple_test
 class SanityFailureCheck(BaseFrontendCheck):
+    '''A performance test that fails in sanity.'''
     @sanity_function
     def validate_output(self):
         return sn.assert_found('foo', self.stdout)
 
+    @performance_function('Gflop/s')
+    def perf(self):
+        return sn.extractsingle(r'perf: (\d+)', self.stdout, 1, int)
```

Comment:
Converting this test to a performance test that fails in sanity triggers the bug with the current unit tests.